images/virtualization-artifact/pkg/controller/service/vm_operation.go

```diff
@@ -188,21 +207,37 @@ func (s VMOperationService) IsComplete(ctx context.Context, vmop *virtv2.Virtual
 			return false, err
 		}
 
-		// Use vmop creation time or time from SignalSent condition.
-		signalSentTime := vmop.GetCreationTimestamp().Time
-		signalSendCond, found := GetCondition(vmopcondition.SignalSentType.String(), vmop.Status.Conditions)
-		if found && signalSendCond.LastTransitionTime.After(signalSentTime) {
-			signalSentTime = signalSendCond.LastTransitionTime.Time
+		return vmPhase == virtv2.MachineRunning &&
+			s.afterSignalSent(kvvmi.GetCreationTimestamp().Time, vmop), nil
+	case virtv2.VMOPTypeMigrate:
+		kvvmi, err := s.getKVVMI(ctx, vmop.GetNamespace(), vmop.Spec.VirtualMachine)
+		if err != nil {
+			return false, err
 		}
 
-		return vmPhase == virtv2.MachineRunning &&
-			kvvmi.GetCreationTimestamp().After(signalSentTime), nil
+		if s.afterSignalSent(kvvmi.GetCreationTimestamp().Time, vmop) {
+			return true, nil
+		}
+		migrationState := kvvmi.Status.MigrationState
+		return migrationState != nil &&
+			migrationState.EndTimestamp != nil &&
+			s.afterSignalSent(migrationState.EndTimestamp.Time, vmop), nil
 
 	default:
 		return false, nil
 	}
 }
 
+func (s VMOperationService) afterSignalSent(timestamp time.Time, vmop *virtv2.VirtualMachineOperation) bool {
+	// Use vmop creation time or time from SignalSent condition.
+	signalSentTime := vmop.GetCreationTimestamp().Time
+	signalSendCond, found := GetCondition(vmopcondition.SignalSentType.String(), vmop.Status.Conditions)
+	if found && signalSendCond.LastTransitionTime.After(signalSentTime) {
```

Comment:
What if signal is not sent yet? signalSendCond is Unknown or False

Reply:
Not an accurate name. We need to check for newly created KVVMI.

Reply:
`IsComplete` method is called only when VMOP is in the InProgress phase, i.e. after sending the signal in operation handler. Using "vmop.creationTimestamp" is a last resort just in case. 

https://github.com/deckhouse/virtualization/blob/13f6a9d425b12b0398341918eabb792f7e15ab8b/images/virtualization-artifact/pkg/controller/vmop/internal/lifecycle.go#L111-L110

Reply:
rename to isAfterSignalSentOrCreation